inference-engine/src/mkldnn_plugin/nodes/region_yolo.cpp

```diff
@@ -173,47 +240,62 @@ class RegionYoloImpl: public ExtLayerBase {
             if (layer->insData.size() != 1 || layer->outData.empty())
                 THROW_IE_EXCEPTION << "Incorrect number of input/output edges!";
 
+            input_prec = layer->insData.front().lock()->getPrecision();
+            output_prec = layer->outData.front()->getPrecision();
+
+            if (input_prec != Precision::FP32 && input_prec != Precision::BF16) {
+                input_prec = Precision::FP32;
+            }
+
+            if (output_prec != Precision::FP32 && output_prec != Precision::BF16) {
+                output_prec = Precision::FP32;
+            }
+
+            if (Precision::BF16 == output_prec) {
+                if (!mayiuse(avx512_core_bf16)) {
+                    output_prec = Precision::FP32;
+                }
+            }
+
             classes = layer->GetParamAsInt("classes");
             coords = layer->GetParamAsInt("coords");
             num = layer->GetParamAsInt("num");
             do_softmax = layer->GetParamAsBool("do_softmax", true);
             mask = layer->GetParamAsInts("mask", {});
 
+            jit_logistic_config_params jcp;
+            jcp.src_dt = jcp.dst_dt = MKLDNNExtensionUtils::IEPrecisionToDataType(output_prec);
+            jcp.src_data_size = jcp.dst_data_size = output_prec.size();
```

Comment:
Why do we limit support by equal precisions on input and output? Jit code can easily mix the up.

Reply:
In this particular context, we use the same data blob as input and output of the jit kernel (because of the problem definition), so we use the same precision as well.